### ISSUE TITLE
Learn testing in Android

### ISSUE THREAD

**Issue by YigitSekerci** [bounswe/bounswe2023group4#220]
### Problem Definition
Testing is maybe the most important part of the apps. Every member must learn how to write testable code and how to test their code via unit test and integration tests.

### Acceptance Criteria
* Learn how to write testable code.
* Learn how to write unit tests in android.
* Learn how to write integration tests in android.

**Comment by YigitSekerci** [comment]
I have followed the [official android testing guides](https://developer.android.com/training/testing/fundamentals) to learn how test mobile apps.
I have started with unit tests and implemented examples for easy adaptation in code.
I have continued with ui test for jetpack compose. At first, I couldn't understand how to write testable jetpack compose UI but I guess I feel more comfortable right now. However, I need to test these principles in complex UI to fully test myself. 
I have learned how to mock api calls and repositories to make tests in a more accurate manner.

In short, I believe I can write both testable code and tests but we will see my performance while coding the project since sometimes some concepts can be different in practice.

**Comment by AliAlperenSonmez** [comment]
I've grasped the significance of testing in app development, recognizing it as a vital aspect of ensuring software quality. I am proficient in writing testable code, enabling me to create code that is easily testable, a crucial skill in software development. I have acquired the expertise to write both unit tests and integration tests specifically for Android, enhancing my ability to thoroughly test and validate my code within the Android ecosystem.

**Comment by YigitSekerci** [comment]
We made a mistake by combining research issues. Because every members' finish time is different we couldn't close this issue on time. Due to these reason, this issue is completed one day after its deadline.
